File: UtilitiesVB/Common Outlook Functions/CreateCategory.vb
Imports Microsoft.Office.Interop
Imports Microsoft.Office.Interop.Outlook

Public Module CreateCategoryModule
    Public Function CreateCategory(OlNS As Outlook.NameSpace, prefix As IPrefix, newCatName As String) As Category

        Dim objCategory As Category = Nothing
        'Dim OlColor As OlCategoryColor
        Dim strTemp As String

        If newCatName <> "" Then
            If prefix.Value <> "" Then
                strTemp = If(Len(newCatName) > Len(prefix),
                    If(Left(newCatName, Len(prefix.Value)) <> prefix.Value, prefix.Value & newCatName, newCatName),
                    prefix.Value & newCatName)
            Else
                strTemp = newCatName
            End If


            Dim exists As Boolean = False
            For Each objCategory In OlNS.Categories
                If objCategory.Name = strTemp Then
                    exists = True
                    Dim unused1 = MsgBox("Color category " & strTemp & " already exists. Cannot add a duplicate.")
                    Return objCategory
                End If
            Next objCategory

            If Not exists Then
                Try
                    objCategory = OlNS.Categories.Add(strTemp,
                                                      prefix.Color,
                                                      OlCategoryShortcutKey _
                                                      .olCategoryShortcutKeyNone)
                Catch ex As System.Exception
                    Debug.WriteLine(ex.Message)
                    Debug.WriteLine(ex.StackTrace.ToString())
                End Try
            End If
        Else
            Dim unused = MsgBox("Error: Parameter " & NameOf(newCatName) & " must have a value to create a category.")
        End If

        Return objCategory

    End Function
End Module

File: UtilitiesVB/Common Outlook Functions/CreateCategory.cs
using System.Diagnostics;
using Microsoft.Office.Interop.Outlook;
using Microsoft.VisualBasic;

namespace UtilitiesVB
{

    public static class CreateCategoryModule
    {
        public static Category CreateCategory(NameSpace OlNS, IPrefix prefix, string newCatName)
        {

            Category objCategory = null;
            // Dim OlColor As OlCategoryColor
            string strTemp;

            if (!string.IsNullOrEmpty(newCatName))
            {
                if (!string.IsNullOrEmpty(prefix.Value))
                {
                    strTemp = Strings.Len(newCatName) > Strings.Len(prefix) ? (Strings.Left(newCatName, Strings.Len(prefix.Value)) ?? "") != (prefix.Value ?? "") ? prefix.Value + newCatName : newCatName : prefix.Value + newCatName;
                }
                else
                {
                    strTemp = newCatName;
                }


                bool exists = false;
                foreach (Category currentObjCategory in OlNS.Categories)
                {
                    objCategory = currentObjCategory;
                    if ((objCategory.Name ?? "") == (strTemp ?? ""))
                    {
                        exists = true;
                        var unused1 = Interaction.MsgBox("Color category " + strTemp + " already exists. Cannot add a duplicate.");
                        return objCategory;
                    }
                }

                if (!exists)
                {
                    try
                    {
                        objCategory = OlNS.Categories.Add(strTemp, prefix.Color, OlCategoryShortcutKey.olCategoryShortcutKeyNone);
                    }
                    catch (System.Exception ex)
                    {
                        Debug.WriteLine(ex.Message);
                        Debug.WriteLine(ex.StackTrace.ToString());
                    }
                }
            }
            else
            {
                var unused = Interaction.MsgBox("Error: Parameter " + nameof(newCatName) + " must have a value to create a category.");
            }

            return objCategory;

        }
    }
}